### PR TITLE
Added Docker Dev Environment & updated login error handling

### PR DESCRIPTION
**Pull request by fgarci03**
- Added Docker Dev Environment
- Updated Login & Protected Data error handling
- Updated package.json dependencies
- Fixed Static Validation issues
- 100% Test Coverage

### CODE REVIEW

**Comment by sanbor** [comment]
The title of this PR is "Added Docker Dev Environment". However, in the Readme it doesn't mention this. It is not clear to me if this Docker configs are intended for development or production. Disclaimer: I have very basic Docker understanding. I just played a little bit with Docker compose and Docker.

**Comment by fgarci03** [comment]
@sanbor Thanks for the input. Will update accordingly :+1:

File: yarn.lock [hunk 1]
```diff
@@ -0,0 +1,5963 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (DFilipeS) [inline yarn.lock]: Add the yarn.lock file to the.gitignore.

**Comment by DFilipeS** [comment]
Seems good to me, works pretty nicely. Thank you!

File: docker/postgres/init-user-db.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+set -e
+
+psql -v ON_ERROR_STOP=1 --username "$POSTGRES_USER" <<-EOSQL
```
Comment (fgarci03) [inline docker/postgres/init-user-db.sh]: Just to avoid using the Postgres root user, nothing more...

**Pull request merged**